godot_project/project_workspace/workspace_context/workspace_context.gd

```diff
@@ -1202,14 +1202,33 @@ func snapshot_moment(in_operation_name: String) -> void:
 	# workaround for part of the state for this workspace_context being inside ScriptUtils (this will apply that state)
 	get_editable_structure_contexts()
 	
-	# Apply simulation if the operation modified the structure of the project
-	if not History.is_operation_whitelisted(in_operation_name):
-		apply_simulation_if_running()
-	
 	# apply overdue signals, ensure snapshots contain up to date state
 	_emit_selection_in_structures_changed()
 	_emit_structure_content_changed()
 	
+	# Apply simulation if the operation modified the structure of the project
+	if not History.is_operation_whitelisted(in_operation_name) and is_simulating():
```

Comment:
I think this is one of those rare cases where it's worth to consider the condition order, I would swap it. 
The key for the person who reads this code is to understand it only applies to the case where simulation is involved so I think 
`if is_simulating() and not History.is_operation_whitelisted(in_operation_name):`
will be easier to read

Reply:
Good catch